### PR TITLE
Fix inconsistency in readiness probe.

### PR DESCRIPTION
**Pull request by GDLMadushanka**
Fixes wso2/micro-integrator/issues/2251

### CODE REVIEW

**Comment by Nirothipan** [comment]
There were test cases for this, did you happen to check them too with the fix

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java [hunk 1]
```diff
@@ -78,33 +77,25 @@ public boolean invoke(MessageContext synCtx) {
...
-            // update cached response
-            cachedResponse = response;
+        ArrayList<String> faultyCapps = new ArrayList<>(CappDeployer.getFaultyCapps());
+        if (!faultyCapps.isEmpty()) {
+            String faultyList = String.join("\",\"", faultyCapps);
+            response += "\"status\": \"not ready, faulty CAPPs detected\", \"Faulty CAPPs\" : [\"" + faultyList
+                    + "\"]}";
+            responseCode = 500;
+        } else {
+            response += "\"status\" : \"ready\"}";
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java:92]: how do we ensure that deployment is completed by this time

Comment (GDLMadushanka) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java:92]: > There were test cases for this, did you happen to check them too with the fix

Removed the invalid tests

Comment (GDLMadushanka) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java:92]: > how do we ensure that deployment is completed by this time

https://ei.docs.wso2.com/en/7.2.0/micro-integrator/setup/deployment/health_check/
quote "Initial delay and the period has to be fine-tuned according to your deployment."

**Review by Nirothipan — APPROVED** [review]

**Pull request merged**